core/lib/health_check/Cargo.toml

```diff
@@ -20,6 +20,7 @@ serde_json.workspace = true
 thiserror.workspace = true
 tokio = { workspace = true, features = ["sync", "time"] }
 tracing.workspace = true
+zksync_bin_metadata.workspace = true
```

Comment:
This dependency looks out of place; it makes a more-or-less isolated library depend on a library with a build script etc. An inverse dependency (`bin_metadata` depending on `health_check`) would make more sense.

Reply:
I see your point here. The problem with inverting the dependency is that I wouldn't be able to add `BIN_METADATA` at the moment I create the inner `Health` instance:

```rust
let inner = Health::with_details(aggregated_status.into(), BIN_METADATA);
```

WDYT?